sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample04_BasicCheckpointing.cs

```diff
@@ -121,37 +121,47 @@ await using (var producer = new EventHubProducerClient(eventHubsConnectionString
 
             async Task processEventHandler(ProcessEventArgs eventArgs)
             {
-               try
-               {
-                   ++eventIndex;
-                   ++eventsSinceLastCheckpoint;
-
-                   if (eventsSinceLastCheckpoint >= eventsPerBatch)
-                   {
-                       await eventArgs.UpdateCheckpointAsync();
-
-                       Console.WriteLine("Created checkpoint");
-                       Console.WriteLine();
-                       eventsSinceLastCheckpoint = 0;
-                   }
-
-                   Console.WriteLine($"Event Received: { Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray()) }");
-               }
-               catch (Exception ex)
-               {
-                   // For real-world scenarios, you should take action appropriate to your application.  For our example, we'll just log
-                   // the exception to the console.
-
-                   Console.WriteLine();
-                   Console.WriteLine($"An error was observed while processing events.  Message: { ex.Message }");
-                   Console.WriteLine();
-               }
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                try
+                {
+                    ++eventIndex;
+                    ++eventsSinceLastCheckpoint;
+
+                    if (eventsSinceLastCheckpoint >= eventsPerBatch)
+                    {
+                        await eventArgs.UpdateCheckpointAsync(eventArgs.CancellationToken);
```

Comment:
That's a fair point.   My intent was to demonstrate that there was a cancellation token involved, but considering the larger scenario as you paint it.... you're right in that it doesn't make sense to use that particular one. 

I'll tweak to use a dedicated one.

Reply:
I noticed `.ConfigureAwait(false)` isn't used. Do we only expect these samples to be used in console applications? I can imagine someone wanting to run this as a background task in their web worker. Just a question. :)

Reply:
`ConfigureAwait` is one of those things that is highly misunderstood in my experience.  It's use is also orthogonal to the sample, for the most part.   

Not having it will potentially degrade performance due to waiting for an unused context to be restored, but won't otherwise cause issue.   Including it and then having the sample code copied to a place where the synchronization context is important would potentially blow up the universe in ways that may not be apparent and may lead to the conclusion that "Event Hubs just doesn't work in an WPF application" or similar.